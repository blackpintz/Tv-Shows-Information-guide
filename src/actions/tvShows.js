const addTvShowData = data => ({
  type: 'ADD_TVSHOW_DATA',
  data,
});

export default addTvShowData;
