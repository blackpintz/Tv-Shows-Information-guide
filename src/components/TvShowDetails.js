import React, { useEffect } from 'react';
import propTypes from 'prop-types';
import axios from 'axios';
import { connect } from 'react-redux';
import { Container, Button } from 'react-bootstrap';
import addTvShowData from '../actions/tvShows';

export const TvShowDetails = ({ onFetch, tvShow, routeProps }) => {
  const { show } = tvShow;
  const { history } = routeProps;
  useEffect(async () => {
    const result = await axios.get('https://api.tvmaze.com/schedule?country=US&date=2020-12-01');
    const { data } = result;
    if (show === 'no value') onFetch(data);
  }, []);
  return (
    <>
      <Container className="d-flex flex-column align-items-center">
        <h1>{show.name}</h1>
        <div className="d-flex">
          <p>{`${show.runtime} mins`}</p>
          {show.genres.length !== 0 ? (
            <div className="d-flex ml-2">
              |
              {show.genres.map(genre => <p key={genre} className="px-1">{`${genre},`}</p>)}
              |
            </div>
          ) : (
            <span className="ml-2">|</span>
          )}
          <div className="d-flex">
            <p className="px-2">{show.type}</p>
            |
          </div>
        </div>
        <div>
          <img src={show.image.original} className="detail-img" alt="show-img" />
          <p className="mt-2 text-left">{`The show has ${tvShow.number} episodes. Watch it every ${show.schedule.days[0]}, at ${show.schedule.time}hours.`}</p>
        </div>
        <Button className="mb-3" onClick={() => history.goBack()}>Go to Homepage</Button>
      </Container>
    </>
  );
};

const mapStateToProps = ({ tvShows }, props) => ({
  tvShow: tvShows.find(show => show.id.toString() === props.id),
});

const mapDispatchToProps = dispatch => ({
  onFetch: data => dispatch(addTvShowData(data)),
});

TvShowDetails.propTypes = {
  tvShow: propTypes.objectOf(propTypes.oneOfType([
    propTypes.string,
    propTypes.number,
    propTypes.object,
  ])),
  routeProps: propTypes.objectOf(propTypes.object),
  onFetch: propTypes.func.isRequired,
  history: propTypes.objectOf(propTypes.func.isRequired),
};

TvShowDetails.defaultProps = {
  tvShow: { show: 'no value' },
  routeProps: { history: 'no value' },
  history: {},
};

export default connect(mapStateToProps, mapDispatchToProps)(TvShowDetails);
