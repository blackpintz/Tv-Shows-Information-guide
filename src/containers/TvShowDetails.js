import React, { useState, useEffect } from 'react';
import propTypes from 'prop-types';
import axios from 'axios';
import { connect } from 'react-redux';
import { Container, Button } from 'react-bootstrap';
import addTvShowData from '../actions/tvShows';

export const TvShowDetails = ({
  onFetch, tvShow, routeProps,
}) => {
  const [loading, setLoading] = useState(false);
  const { history } = routeProps;
  useEffect(async () => {
    const result = await axios.get('https://api.tvmaze.com/schedule?country=US&date=2020-12-01');
    const { data } = result;
    onFetch(data);
    setLoading(true);
  }, []);
  const { show } = tvShow;
  return (
    <>
      { loading ? (
        <Container className="d-flex flex-column align-items-center">
          <h1>{show.name}</h1>
          <div className="d-flex">
            <p>{`${show.runtime} mins`}</p>
            {show.genres.length !== 0 ? (
              <div className="d-flex ml-2">
                |
                {show.genres.map(genre => <p key={genre} className="px-1">{`${genre},`}</p>)}
                |
              </div>
            ) : (
              <span className="ml-2">|</span>
            )}
            <div className="d-flex">
              <p className="px-2">{show.type}</p>
              |
            </div>
          </div>
          <div>
            <img src={show.image.original} className="detail-img" alt="show-img" />
            <p className="mt-2 text-left">{`The show has ${tvShow.number} episodes. Watch it every ${show.schedule.days[0]}, at ${show.schedule.time}hours.`}</p>
          </div>
          <Button className="mb-3" onClick={() => history.goBack()}>Go to Homepage</Button>
        </Container>
      ) : (<></>)}
    </>
  );
};

const mapStateToProps = ({ tvShows }, props) => ({
  tvShow: tvShows.find(show => show.id.toString() === props.id),
});

const mapDispatchToProps = dispatch => ({
  onFetch: data => dispatch(addTvShowData(data)),
});

TvShowDetails.propTypes = {
  tvShow: propTypes.objectOf(propTypes.oneOfType([
    propTypes.string,
    propTypes.number,
    propTypes.object,
  ])),
  onFetch: propTypes.func.isRequired,
  routeProps: propTypes.objectOf(propTypes.object),
  history: propTypes.objectOf(propTypes.func.isRequired),
};

TvShowDetails.defaultProps = {
  tvShow: { show: 'no value' },
  routeProps: { history: 'no value' },
  history: {},
};

export default connect(mapStateToProps, mapDispatchToProps)(TvShowDetails);
