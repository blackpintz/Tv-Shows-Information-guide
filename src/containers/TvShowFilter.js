import React from 'react';
import propTypes from 'prop-types';
import Select from 'react-select';
import { connect } from 'react-redux';
import showGenres from '../constants';
import changeFilter from '../actions/filter';

export const TvShowFilter = ({ onFilter }) => (
  <Select
    options={showGenres}
    onChange={val => onFilter(val.label)}
    placeholder="Choose Genre"
    className="mb-3"
  />
);

const mapStateToProps = ({ filter }) => ({
  filter,
});

const mapDispatchToProps = dispatch => ({
  onFilter: val => dispatch(changeFilter(val)),
});

TvShowFilter.propTypes = {
  onFilter: propTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(TvShowFilter);
