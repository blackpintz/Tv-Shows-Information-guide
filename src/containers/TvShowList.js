import React, { useEffect } from 'react';
import axios from 'axios';
import { connect } from 'react-redux';
import propTypes from 'prop-types';
import { Container, Row } from 'react-bootstrap';
import TvShowListItem from '../components/TvShowListItem';
import addTvShowData from '../actions/tvShows';
import ConnectTvShowFilter from './TvShowFilter';
import filteredtvShows from '../selectors/filtertvShows';

const TvShowList = ({ onFetch, tvShows }) => {
  useEffect(async () => {
    const result = await axios.get('https://api.tvmaze.com/schedule?country=US&date=2020-12-01');
    const { data } = result;
    onFetch(data);
  }, []);
  return (
    <>
      <Container>
        <h1>TV Information Guide.</h1>
        <ConnectTvShowFilter />
        <Row>
          {tvShows.map(show => <TvShowListItem key={show.id} show={show} />)}
        </Row>
      </Container>
    </>
  );
};

const mapStateToProps = ({ tvShows, filter }) => ({
  tvShows: filteredtvShows(tvShows, filter),
});

const mapDispatchToProps = dispatch => ({
  onFetch: data => dispatch(addTvShowData(data)),
});

TvShowList.propTypes = {
  onFetch: propTypes.func.isRequired,
  tvShows: propTypes.arrayOf(propTypes.object),
};

TvShowList.defaultProps = {
  tvShows: 'no value',
};

const connectTvShowList = connect(mapStateToProps, mapDispatchToProps)(TvShowList);

export { connectTvShowList as default, TvShowList };
