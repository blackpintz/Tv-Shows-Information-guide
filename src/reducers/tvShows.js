const tvShowsReducer = (state = [], action) => {
  switch (action.type) {
    case 'ADD_TVSHOW_DATA':
      return [
        ...state, action.data,
      ];
    default:
      return state;
  }
};

export default tvShowsReducer;
