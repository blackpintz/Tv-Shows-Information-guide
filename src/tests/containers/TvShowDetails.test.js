import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { TvShowDetails } from '../../containers/TvShowDetails';
import tvShows from '../fixtures/tvShows';

Enzyme.configure({ adapter: new Adapter() });

test('should render TvShowList component with the list of tvShows', () => {
  const onFetch = jest.fn();
  const rProps = {
    history: {
      goBack: jest.fn(),
    },
  };
  const wrap = shallow(<TvShowDetails onFetch={onFetch} tvShow={tvShows[0]} routeProps={rProps} />);
  expect(wrap).toMatchSnapshot();
});
