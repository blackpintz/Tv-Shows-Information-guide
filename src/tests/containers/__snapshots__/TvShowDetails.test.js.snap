// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TvShowList component with the list of tvShows 1`] = `<Fragment />`;
